app/jobs/precompute_student_hashes_job.rb

```diff
@@ -34,7 +34,8 @@ def precompute_and_write_student_hashes!(date_timestamp, authorized_student_ids)
     key = precomputed_student_hashes_key(precomputed_time, authorized_student_ids)
 
     # This is a non-atomic upsert
-    PrecomputedQueryDoc.find(key).destroy! if PrecomputedQueryDoc.exists?(key)
+    pre_existing_doc = PrecomputedQueryDoc.find_by_key(key)
```

Comment:
@kevinrobinson Of course! Thank you for writing all the precompute logic, it is awesome and speeds up the overview soooo much as I shall document shortly! 

😊